anchor-cli.nix

```diff
@@ -5,14 +5,23 @@
 , libgcc
 , pkg-config
 , protobuf
-, rustPlatform
+, makeRustPlatform
 , makeWrapper
 , solana-platform-tools
+, rust-bin
 , udev
 }:
+let
+  # Anchor does not declare a rust-toolchain, so we have to do it here -- the
+  # old dependency on the `time` crate doesn't support Rust versions >= 1.80.
+  rustPlatform = makeRustPlatform {
+    cargo = rust-bin.stable."1.79.0".default;
+    rustc = rust-bin.stable."1.79.0".default;
+  };
```

Comment:
This causes a bunch of packages to be built from source, although not a major issue as it only takes a few mins, but is there any way to get a rustPlatform of different versions from the binary cache?

Reply:
nix only bundles one version of the toolchain, per https://nixos.wiki/wiki/Rust#Installing_via_nixpkgs, and the docs recommend using either this overlay or the mozilla overlay if you want to use another toolchain https://nixos.wiki/wiki/Rust#Custom_Rust_version, which makes me think that there might not be an easier way to do this, outside of using an older version of nixpkgs.

I also tried using a local rust-toolchain.toml and patching one into the Anchor repo, but neither one worked.

As I mentioned though, I'm still new to nix, so there might certainly be a better way!